experimental/envoy_proxy/README.md

```diff
@@ -0,0 +1,68 @@
+# Envoy Proxy Experiment
+
+This experiment shows how to create a continuous connection to a Docker container in Cloud Run. It is done by
+using [TCP tunneling over HTTP](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/http/upgrades#tunneling-tcp-over-http) in Envoy proxy.
+
+The experiment consists of 2 Envoy proxies: _client side_ and _server side_.
+- Client side proxy accepts TCP streams and tunnels them over HTTP streams.
+- Server side proxy listens for HTTP streams and unwraps TCP streams from them.
+
+Proxy experiment is organized as follows:
+- `curl`
+  - Sends an HTTP GET request on `localhost:8000`
+- `client_listener`
+  - Tunnels TCP stream over HTTP/2 stream
+- `client_cluster`
+  - Sends HTTP/2 stream on `APP_URL:443` using TLS
+  - Checks Google public certificate
+- Cloud Run Frontend
```

Comment:
I think there is another component here which is the Cloud LB, or is it the Frontend? Basically there is something that keeps the connection open, and sends individual requests to Cloud Run instances, over individual connections.

Reply:
As Cloud Run team pointed out, it looks like this:
client -> raw TCP (localhost) -> client envoy TCP proxy listener -> (tunnel TCP stream over HTTP/2 stream with TLS (Google public cert)) -> Cloud Run Frontend (TLS terminated) -> plaintext HTTP/2 (h2c) -> server side HTTP envoy -> (extract TCP stream from H2C stream) -> raw TCP (localhost) -> server

So Cloud Frontend forwards HTTP streams to Docker containers